### DISCUSSION TITLE
getltscfg replacement (get custom params from ltsp.conf on client)

### DISCUSSION THREAD

**Discussion by hubeny** [ltsp/ltsp#502] [Q&A] [answered]
Hi,

in old LTSP5 evironment we used on thin clients command 

`getltscfg PARAM_NAME`

this command returns the value of the `PARAM_NAME` which was defined on server in the file `/var/lib/tftpboot/ltsp/lts.conf`.

Example:
In the file `/var/lib/tftpboot/ltsp/lts.conf` on the server we have:

```
[Default]
BARCODE_READER_RS232_PORT = 0
...

[tc17]
BARCODE_READER_RS232_PORT = 97
...
```

when the client `tc17` boots, the start scripts calls `getltscfg BARCODE_READER_RS232_PORT` and we have known that the barcode reader is connected to virtual port `/dev/ttyS97` on this particular client  and can use this value in our start scripts. 

Which mechanism can be used in current LTSP version (with config file in `/etc/ltsp/ltsp.conf` on the server). 

In [https://github.com/ltsp/ltsp/discussions/268](https://github.com/ltsp/ltsp/discussions/268) @alkisg announces, that getltscfp is replaced by an internal shell function. However, We did not find in the available documentation, how to achieve the same goal as described above.

Can anybody help with it?

Thank you in advance,

Jan Hubený

**Comment by alkisg** [comment, marked as answer]
All the ltsp.conf variables are internal to LTSP scripts. They may be different in each applet (e.g. `ltsp init` may have different variables than `ltsp login` even on the same client).
To save variables for external programs, you could do something like:

```shell
[tc17]
POST_INIT_CREATE_CONFIG="echo 'BARCODE_READER_RS232_PORT = 97' >/etc/ltsp/external-program.conf"
```

Then you can source the generated conf file from your external program.

Where are you running your external program from? If you run it from an LTSP hook like POST_INIT, you can just pass the variable as a parameter since you're already "inside" LTSP.

Finally in some cases it might be convenient to write custom LTSP applets in /usr/share/ltsp/client/your-applet. Then you could run them with `ltsp your-applet` and they'd have full access to LTSP variables.

**Reply by hubeny** [reply]
Thank you for fast reply. 

Our external program is bash script, which either starts barcode reader server (this server reads all values on RS232 and send this values to registered apps via TCP/IP).

As you write, it will be convenient for us to run the script directly with LTSP hook. This is evident solution. Thank you for your help.
